### ISSUE TITLE
TypeError improperly raised when using nested Literal types

### ISSUE THREAD

**Issue by rfrowe** [agronholm/typeguard#128]
## Problem

When using a `Literal` type nested within `Union`, the wrong `TypeError` is raised on illegal values.

## Reproduction

```python3
>>> from typeguard import check_type
>>> from typing import Literal, Union
>>> check_type('foo', 'foo', Union[float, Literal['bar'])
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/typeguard/__init__.py", line 668, in check_argument_types
    check_type(description, value, expected_type, memo)
  File "/usr/local/lib/python3.8/dist-packages/typeguard/__init__.py", line 566, in check_type
    checker_func(argname, value, expected_type, memo)
  File "/usr/local/lib/python3.8/dist-packages/typeguard/__init__.py", line 408, in check_union
    typelist = ', '.join(get_type_name(t) for t in union_params)
TypeError: sequence item 1: expected str instance, NoneType found
```

## Expected Behavior

`TypeError: type of foo must be one of (float, Literal['bar']); got 'foo' instead`

or

`TypeError: type of foo must be one of (float, 'bar'); got 'foo' instead`

## Cause

It seems `Literal` defines neither `__name__` nor `_name` so `get_type_name` returns `None`, which then raises a `TypeError` during `str.join`

**Comment by agronholm** [comment]
Fixed via c79bd2e0b88fdd497322e247c880688d5a94e481.

**Comment by rfrowe** [comment]
@agronholm Thanks!
